app/controllers/account_controller.rb

```diff
@@ -115,11 +115,25 @@ def register
 
     if request.get?
       session[:auth_source_registration] = nil
-      @user = User.new(language: Setting.default_language)
+
+      if session.include? :invitation_token
+        token = Token.find_by(value: session[:invitation_token])
+
+        @user = token.user
```

Comment:
You have the same code for building the user from token in both blocks.
Why not move that out of the request condition?


``` ruby
if session.include? :invitation_token
  ...
elsif request.get?
  ...
else
  ... 
end